src/gui/gui_element_types.hpp

```diff
@@ -331,8 +331,12 @@ class generic_tabbed_window : public main_window_element_base {
 class generic_close_button : public button_element_base {
 public:
 	void button_action(sys::state& state) noexcept override {
-		if(parent)
+		if(parent) {
 			parent->set_visible(state, false);
+			if(!state.ui_state.can_move_map_while_visible) {
+				state.ui_state.can_move_map_while_visible = true;
```

Comment:
This button is used in a lot of places. For example, it is used in the build factory subwindow. This would mean that if you are in the production window, and then open and close the build-a-factory window, that you would be able to move the map again while in the production window

Reply:
(to be clear, if you *cancelled* out of building)